server/src/main/java/io/druid/metadata/storage/derby/DerbyEmbeddedMetadataStorageDruidModule.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.metadata.storage.derby;
+
+import com.google.inject.Binder;
+import com.google.inject.Key;
+import io.druid.guice.LazySingleton;
+import io.druid.guice.PolyBind;
+import io.druid.guice.SQLMetadataStorageDruidModule;
+import io.druid.metadata.MetadataStorageConnector;
+import io.druid.metadata.MetadataStorageProvider;
+import io.druid.metadata.SQLMetadataConnector;
+
+/**
+ * Most useful for unit testing when `org.apache.derby.jdbc.EmbeddedDriver` is needed
+ */
+public class DerbyEmbeddedMetadataStorageDruidModule extends SQLMetadataStorageDruidModule
```

Comment:
`DerbyEmbeddedConnector` uses `org.apache.derby.jdbc.EmbeddedDriver` instead of the client driver. Another alternative is to have a setting that allows the configuration of the driver class.